libraries/AP_AHRS/AP_AHRS_NavEKF.cpp

```diff
@@ -1743,29 +1751,26 @@ uint8_t AP_AHRS_NavEKF::get_primary_IMU_index() const
     return imu;
 }
 
-// get earth-frame accel vector for primary IMU
-const Vector3f &AP_AHRS_NavEKF::get_accel_ef() const
-{
-    return get_accel_ef(get_primary_accel_index());
-}
-
-
-// get the index of the current primary accelerometer sensor
-uint8_t AP_AHRS_NavEKF::get_primary_accel_index(void) const
-{
-    if (ekf_type() != 0) {
-        return get_primary_IMU_index();
-    }
-    return AP::ins().get_primary_accel();
-}
-
 // get the index of the current primary gyro sensor
 uint8_t AP_AHRS_NavEKF::get_primary_gyro_index(void) const
 {
-    if (ekf_type() != 0) {
-        return get_primary_IMU_index();
+    int8_t imu = -1;
+    switch (ekf_type()) {
```

Comment:
should this be active_EKF_type() ? ie. if we are failed onto DCM in plane, then I think we shouldn't be asking the failed EKF which IMU to use

Reply:
Maybe, I didn't modify the function I copied.